### PR TITLE
LAW batons- sec stun rework

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
full design doc: https://docs.google.com/document/d/14plimZ_ErM0POPbus_CeX1-2AQD1cSD9oUdWZM15RKQ/edit#heading=h.plx3abia0o5c
Adds LAW batons, a smart baton for a modern age. These batons  are fully modular- they can be given modifications ,similar to those available for a  PKA. Their capacitors and micro lasers can be removed and upgraded, upgrading their lethal and nonlethal modes respectively, though at the cost of using mod capacity
some LAWbaton mods change the baton's emitter. These come with a new baton sprite and generally change the baton's playstyle significantly
LAW batons have a lethal, and a nonlethal mode. Their lethal mode cannot be used on targets that are immune to stun-based attacks, unless someone with armory access uses their ID to remove this restriction, which takes significant enough time that a shitter cannot easiler hand it out to officers.  Their nonlethal mode also stuns silicons, though not as effectively as a flash would.
LAW batons do not have removable power cells. each attack uses about 5% power, not accounting for damage changes or other modifications. Attacks on targets in crit or stamcrit use more power, to discourage the use of batons to crit-then-kill, or chain-stun targets. 
LAW batons must be registered to a user with a weapons permit before they will turn on. This is as simple as swiping an ID, and enables their onboard GPS system. They may be reset with a multitool to disable their gps and unregister the baton, but this renders it unusable until registered again. While being registered doesnt prevent others from using the baton, it alllows thieves to be tracked down- the LAWbaton is meant to be a weapon an officer takes care of and builds up over the course of a round, so its loss is significant

on flashes:
Flashes (and to a lesser extent flashbangs) have had their combat niche SIGNIFICANTLY changed.  They no longer function as a tool that can win a fight with one click. Instead, they are either a self defense tool, or a method of gaining a major advantage for an encounter. When using a flash, the target will be stunned, though the stun is shorter than that of a shove-kick. This also applies an amount of confusion and blindness. The target is also inflicted with a 30 second debuff. This debuff  applies a decent amouunt of slowdown, but the target is immune to further stuns from flashes for the duration- instead, flashing refreshes the debuff and applies more confusion/obscuring/blindness.

People with flash protection are still immune to this debuff. However, now that flashes are far less significant in combat, powergaming flash protection will, hopefully, be less common

- [x] Add LAW batons and all of their bells and whistles
- [x] Rebalance flashes and flashbangs
- [x] Rebalance stunbatons to be on the level of lawbatons, other balance changes such as nerfing sec underclothes (underclothes shouldnt have more than 10 armor holy shit why did sec jumpsuits and hailers have 30 stamina armor)
- [x] Fix the stupid stamcrit delay bug
- [ ] fully implement LAW batons- put them in vendors, armory, seccie inventories, cargo, etc
- [ ] add a good deal of mods and customizeability for LAW batons to work with
- [ ] fully implement LAW baton mods, adding them to various places where they can be found and such. Feel free to make suggestions!
- [ ] get a spriter to make real sprites god my codersprites suck. If they want to redesign them away from being weird laser swords, go hog wild.

## Why It's Good For The Game
Currently, security has a very powerful melee kit that is incredibly oppressive to melee combat as a whole. They also have no progression throughout the game, no way to effectively deal with simplemobs (even opening the armory has limited efficacy). You don't even stand a chance in station combat unless using flash protection. I want to give sec a more fun to use kit that's healthier for melee combat as a whole. 

This also solves the issue of security being absolutely useless when a simplemob threat occurs on-station. Giving your average beat cop a way to fight some of the more dangerous and pressing threats other than whaling on it with a 5 damage harmbaton is, overall, a major boon

## Changelog
:cl:
add: LAWbatons! A modular stun baton replacement. They have parts which can be replaced to modify their damage and other effects, and a lethal mode which will only work on simplemobs in normal situations (unless the target is stun immune, or  someone with armory access has 
tweak: flashes and flashbang behaviour changed. Flashing now has a short stun (not long enough to cuff, but it lasts long enough for a few melee hits, or to run away from a threat), and gives blindness, confusion, and obscures vision. It also applies a 30 second debuff that is refreshed whenever the target is flashed. This debuff gives a firesuit-equivalent slowdown, but the target cannot be stunned by flashes for the duration of the debuff.
balance: stunbatons have been nerfed to be in line with unmodified LAWbatons
balance: sec undersuits and masks have had their stamina armor reduced to 10. There won't be much change when they're fighting stunbatons and the like, due to their armor providing piercing protection
fix: fixes some dumb twohanded component shit
fix: fixes the delay between being put into stamcrit and falling down
imageadd: LAWbaton related sprites
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
I should note that this is an idea that's an experiment and a proof of concept. It'll need to get merged for a different period and go through definite growing pains. if it's not right for the game, I wont get butthurt, but sec has needed major change for years, and I think that if such a change is going to happen, a radical change like this should be given a chance.

**Comment by Naevii** [comment]
How baton works in non lethal mode? Is it similar to how baton is now? Can it be used by pacifists?

**Comment by RDS88-beeman** [comment]
from the design doc, the lethal mode checks if a target is stunnable, and if they are it just doesn't work, so i see no reason a pacifist wouldn't use it. How it works in nonlethal mode is based solely on what head you add

**Comment by zeskorion** [comment]
> How baton works in non lethal mode? Is it similar to how baton is now? Can it be used by pacifists?

Nonlethal mode does 12 × (capacitor rating + 1) stamina damage (24-60), reduced by stamina armor. A piercing protected target counts as having more armor,  though enough armor pen on the baton negates this

Outside of damage and the bells and whistles, it's nearly identical to the stunbaton in jts base form, though its highest damage state is 15 below current stunbaton damage.

The nonlethal mode is usable by pacifists. The lethal mode is not.
Lethal mode damage is 8 + (4*laser rating), if curious

**Comment by TotallyNotChief** [comment]
remember that other baton you added that no one used

also how many "sec reworks" and "stamina reworks" do you really need before you're satisfied?

**Comment by zeskorion** [comment]
> remember that other baton you added that no one used

didnt want to add them
> also how many "sec reworks" and "stamina reworks" do you really need before you're satisfied?

till it's no better than antag nonlethals

**Comment by TotallyNotChief** [comment]
18953779 814999998_image

**Comment by RDS88-beeman** [comment]
> Fix the stupid stamcrit delay bug
 @zeskorion atomize this please i want it NOW

**Comment by zeskorion** [comment]
> > Fix the stupid stamcrit delay bug
> > @zeskorion atomize this please i want it NOW

I did, actually.  Some maints are arguing that said bug is a feature. Its also included in this or because its necessary for the balance of the pr even if it isnt merged solo

**Comment by ItsTisme** [comment]
A issue I wana point out is that "the LAWbaton is meant to be a weapon an officer takes care of and builds up over the course of a round, so its loss is significant" but the fact its ID locked removes the incentive to actuely steal them.  so they rarely will lose it aside from clowns.

Maby make it use similair system as firing pins? where it has to be researched first. round start this is big no no

**Comment by zeskorion** [comment]
> 

The way locking works, if an officer wants to be able to react to a crime, their baton will usually be unlocked, as the process isnt instant. Stealing is only de-incentivized with GPS tracking

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -388,6 +388,30 @@
...
+#define COMSIG_BATON_OVERCHARGING "baton_overcharged" //triggers when a baton gains overcharge
+#define COMSIG_BATON_OVERCHARGE_PLUS "baton_overcharge_plus" //triggers when a baton would overcharge when already overcharged.
+	#define COMSIG_BATON_NOOVERCHARGEMESSAGE 1 //used for modules that have their own effect on exceeding overcharge
+#define COMSIG_BATON_BLOCK "baton_block" //triggers when a baton blocks an attack
+#define COMSIG_BATON_BLOCKED "baton_blocked" //triggers when a baton is blocked by an attack
+	#define COMSIG_BATON_UNBLOCKABLE 1
+#define COMSIG_BATON_FAILED_STUNCHECK "baton_failed_stuncheck" //triggers when a baton hits with an attack against a target, but isn't allowed to attack due to a failed check
+#define COMSIG_BATON_THROWN "baton_thrown" //triggered when a baton is thrown
+#define COMSIG_BATON_THROW_IMPACT "baton_throw_impact" //triggers when a thrown baton hits a target
+#define COMSIG_BATON_EXAMINED "baton_examined" //triggers when a baton is examined
```
Comment (Fikou) [inline code/__DEFINES/dcs/signals.dm:412]: you dont need to create a new signal for everything that the baton does dude just use the regular examine signal

Comment (zeskorion) [reply code/__DEFINES/dcs/signals.dm:412]: i was unaware there was one

**Comment by Howluinb** [comment]
> > 
> 
> The way locking works, if an officer wants to be able to react to a crime, their baton will usually be unlocked, as the process isnt instant. Stealing is only de-incentivized with GPS tracking

I already tried to convince you many, many times that gps tracking a stolen stun batong is an extreme nerf to antag and that saying "well it's only to punish assistants/clowns lol, real antags just use their antag gear" is a terrible idea.

Let alone the antags who don't have gear/lost their gear - i've seen barely any nonantag in months steal an officer's batong or disabler. This isn't a widespread enough problem to warrant this, and especially not something that prevents any antag from fighting back against security with their own weapons.

There's a ton more things wrong with this PR, like how you bundle a flash nerf, the stam crit fix & the new batongs altogether, but only on the fact you refuse to budge on keeping a GPS linked to any batongs at all time - this is bad. and claiming you can just remove the gps part with emagging is just as dumb as not having it removable - a single antag out of our like, dozen antags in the pool has access to easy emagging (excluding ninja whom i believe cant emag in-hand objects anyways).

**Comment by zeskorion** [comment]
> > > 
> > 
> > 
> > The way locking works, if an officer wants to be able to react to a crime, their baton will usually be unlocked, as the process isnt instant. Stealing is only de-incentivized with GPS tracking
> 
> I already tried to convince you many, many times that gps tracking a stolen stun batong is an extreme nerf to antag and that saying "well it's only to punish assistants/clowns lol, real antags just use their antag gear" is a terrible idea.
> 
> Let alone the antags who don't have gear/lost their gear - i've seen barely any nonantag in months steal an officer's batong or disabler. This isn't a widespread enough problem to warrant this, and especially not something that prevents any antag from fighting back against security with their own weapons.
> 
> There's a ton more things wrong with this PR, like how you bundle a flash nerf, the stam crit fix & the new batongs altogether, but only on the fact you refuse to budge on keeping a GPS linked to any batongs at all time - this is bad. and claiming you can just remove the gps part with emagging is just as dumb as not having it removable - a single antag out of our like, dozen antags in the pool has access to easy emagging (excluding ninja whom i believe cant emag in-hand objects anyways).

if the gps functionality turns out to be a huge burden in play, it can be removed.  I'm not very attached to that part of the concept

The stam crit fix is included in a separate PR as well, but it's essential to this one, so i bundle it in in case the maints are slow, and for testing purposes

The flash nerfs are essential to accomplishing the goal of this pr (making stuns and sec gear less oppressive for melee)

**Comment by VictorVow** [comment]
The first issue is bundling flash changes and baton changes into one PR. This should be atomised for that alone. 

**LAWbatons** - idea of an upgradeable baton is cool since sec lacks progression, though I disagree with the GPS tracking/locking as it nerfs antags too hard. You reworked cult swords and gear so that anyone can use them (albeit suffering some damage) by arguing that SS13 is based around the idea that if anyone can obtain an item, they should be able to use it. Therefore antags should still be able to easily use these weapons too.

> balance: stunbatons have been nerfed to be in line with unmodified LAWbatons (from 70 to 24 damage)

This is just bizarre, especially considering stamina itself was recently reworked so that you stop being stunned faster. 

> tweak: flashes and flashbang behaviour changed. Flashing now has a short stun (not long enough to cuff, but it lasts long enough for a few melee hits, or to run away from a threat), and gives blindness, confusion, and obscures vision. It also applies a 30 second debuff that is refreshed whenever the target is flashed. This debuff gives a firesuit-equivalent slowdown, but the target cannot be stunned by flashes for the duration of the debuff.

Flashbangs are already crap since they knockdown rather than stun, why does it need further nerfing? 

And flashes themselves could use a rework, but this doesn't fit. If the stun gets changed to being less than half the duration (7s -> 3s), then why do multiple applications need to make it zero duration? If the stun is that short anyway then it should at least consistently have the same effect.

**Comment by zeskorion** [comment]
> The first issue is bundling flash changes and baton changes into one PR. This should be atomised for that alone.
> 
> **LAWbatons** - idea of an upgradeable baton is cool since sec lacks progression, though I disagree with the GPS tracking/locking as it nerfs antags too hard. You reworked cult swords and gear so that anyone can use them (albeit suffering some damage) by arguing that SS13 is based around the idea that if anyone can obtain an item, they should be able to use it. Therefore antags should still be able to easily use these weapons too.
> 
> > balance: stunbatons have been nerfed to be in line with unmodified LAWbatons (from 70 to 24 damage)
> 
> This is just bizarre, especially considering stamina itself was recently reworked so that you stop being stunned faster.
> 
> > tweak: flashes and flashbang behaviour changed. Flashing now has a short stun (not long enough to cuff, but it lasts long enough for a few melee hits, or to run away from a threat), and gives blindness, confusion, and obscures vision. It also applies a 30 second debuff that is refreshed whenever the target is flashed. This debuff gives a firesuit-equivalent slowdown, but the target cannot be stunned by flashes for the duration of the debuff.
> 
> Flashbangs are already crap since they knockdown rather than stun, why does it need further nerfing?
> 
> And flashes themselves could use a rework, but this doesn't fit. If the stun gets changed to being less than half the duration (7s -> 3s), then why do multiple applications need to make it zero duration? If the stun is that short anyway then it should at least consistently have the same effect.

> atomization

These changes are meant to work with one another. I'm not atomizing them, as the desired effect can only be achieved with all of them working in tandem


> lawbaton gps 

I wanna see how it plays out. If it's poorly recieved i will remove it 

> stunbatons 

The primary idea of this pr is a stunbaton nerf. Lawbatons can become somewhere near what stunbatons are now, but not quite. Stunbatons have insane damage and their previous nerfs werent enough. Stunbatons are strictly inferior to lawbatons as of this pr


> flash

You ignored everything i said. Flashes are no longer a win condition. They apply a hefty slowdown instead.  Flashbangs do what flashes do, for consistency

**Comment by Terra1** [comment]
Ah, brilliant; more soul-destroying stupidity from coders worthy of being a /tg/ headcoder.

> LAW batons  

That is the most boring, unimaginative name anyone could possibly think of. Congratulations on your accomplishment.

> they can be given modifications ,similar to those available for a PKA  

The PKA is quite possibly the worst model for implementing modularity; it's boring, basically pointless other than a few mods, and completely unused - a combination of the crusher and plasma cutter is just objectively better than _any_ PKA, regardless of the modules.

Also, this gamification of a core security feature is entirely unnecessary, stupid, and, frankly, insulting. The only thing in Space Station 13 which should be and remain this gamified is mining and the goddamned arcades, not the department whose literal job is to shut down "fun" "activities" by the crew.

> LAW batons have a lethal, and a nonlethal mode. Their lethal mode cannot be used on targets that are immune to stun-based attacks, unless someone with armory access uses their ID to remove this restriction, which takes significant enough time that a shitter cannot easiler hand it out to officers.  

ID access should be an addon to an existing, unsecured device, not a baked-in "feature" whose sole purpose is to permanently lock out unauthorized users (i.e. implementing a CM-style skill system without actually implementing a CM-style skill system). Any and all restrictions on anything should be possible to bypass with nothing more than your character's bare hands, time, and enough experience with the game (not some arbitrary skill stat, but the actual player's knowledge of the game mechanics).

Also, the harmbaton is _soulful_, and I, along with all other self-respecting shitsec, will immediately switch to baseball bats and toolboxes for arrests if it's removed.

> weapons permit  

Absolutely disgusting. What bombastic, prideful, FUCKING SELF-ASSURED CLUELESS NARCISSISTIC IDI-oh, it's been in the game for ages. Still, it isn't really a used feature besides being used for secbot identification, and it *really* should remain that way. Tying more stuff to IDs is boring, stupid, and wrong, as stated above.

>Flashes (and to a lesser extent flashbangs) have had their combat niche SIGNIFICANTLY changed. They no longer function as a tool that can win a fight with one click.  

Sounds like someone got flashed. Cope.

>This also solves the issue of security being absolutely useless when a simplemob threat occurs on-station.  

There's an easier solution; overhaul simplemobs. They've been basically unchanged for, what, a decade? Perhaps instead of deciding to redesign security to counter shitty, ancient mechanics, redesign those shitty, ancient mechanics?

**Comment by zeskorion** [comment]
> > they can be given modifications ,similar to those available for a PKA
> 
> The PKA is quite possibly the worst model for implementing modularity; it's boring, basically pointless other than a few mods, and completely unused - a combination of the crusher and plasma cutter is just objectively better than _any_ PKA, regardless of the modules.

The only thing i took from the PKA is point-based modspace. Im not using the "haha big mod number" thing.

> Also, this gamification of a core security feature is entirely unnecessary, stupid, and, frankly, insulting. The only thing in Space Station 13 which should be and remain this gamified is mining and the goddamned arcades, not the department whose literal job is to shut down "fun" "activities" by the crew.

Sec can have fun too. You seem absolutely toxic to play with

> 
> > LAW batons have a lethal, and a nonlethal mode. Their lethal mode cannot be used on targets that are immune to stun-based attacks, unless someone with armory access uses their ID to remove this restriction, which takes significant enough time that a shitter cannot easiler hand it out to officers.
> 
> ID access should be an addon to an existing, unsecured device, not a baked-in "feature" whose sole purpose is to permanently lock out unauthorized users (i.e. implementing a CM-style skill system without actually implementing a CM-style skill system). Any and all restrictions on anything should be possible to bypass with nothing more than your character's bare hands, time, and enough experience with the game (not some arbitrary skill stat, but the actual player's knowledge of the game mechanics).

The ID lock mechanic doesnt matter past new lawbatons. It takes too long to lock and unlock to do it every time you need to bring a baton out

> Also, the harmbaton is _soulful_, and I, along with all other self-respecting shitsec, will immediately switch to baseball bats and toolboxes for arrests if it's removed.

Holy shit you're the worst kind of player. Your opinion is going out the window

> > weapons permit
> 
> Absolutely disgusting. What bombastic, prideful, FUCKING SELF-ASSURED CLUELESS NARCISSISTIC IDI-oh, it's been in the game for ages. Still, it isn't really a used feature besides being used for secbot identification, and it _really_ should remain that way. Tying more stuff to IDs is boring, stupid, and wrong, as stated above.
> 

If id becomes an issue i will remove it

> > Flashes (and to a lesser extent flashbangs) have had their combat niche SIGNIFICANTLY changed. They no longer function as a tool that can win a fight with one click.
> 
> Sounds like someone got flashed. Cope.

I play plasmaman.

> > This also solves the issue of security being absolutely useless when a simplemob threat occurs on-station.
> 
> There's an easier solution; overhaul simplemobs. They've been basically unchanged for, what, a decade? Perhaps instead of deciding to redesign security to counter shitty, ancient mechanics, redesign those shitty, ancient mechanics?

Overhauling simplemobs is the opposite of a solution.  As it stands stuncombat sucks anyways, why would i wanna make them susceptible




You're living in the past

**Comment by Terra1** [comment]
>Sec can have fun too. You seem absolutely toxic to play with

Gamification =/= Fun. Standardized security equipment is better than turning it into an absurd mess of a "modular weapons system" loosely based on a mechanic that people pretty much *universally* agree is complete shit (crusher/PKA mods).

>Holy shit you're the worst kind of player.

*best

>I play plasmaman.

Plasmemes can still be flashed by plasmaman security officers in a plasma-filled room when you're not wearing an envirosuit.

> You're living in the past

And you're living on Goon. Shovestun combat - though not perfect - is to date the only combat system that relied more on skill than RNG, and feels fair to both sides in a fight. It also allows for remarkable "underdog" wins, such as an assistant beating a nuclear operative by shoving them into a wall and stealing their gun. It should be kept intact.

**Comment by BrumalLake** [comment]
>Plasmemes can still be flashed by plasmaman security officers in a plasma-filled room when you're not wearing an envirosuit.

this is a joke right

**Comment by zeskorion** [comment]
I've hit a bit of a burnout phase. I'll come back to this project in the future

**Comment by TotallyNotChief** [comment]
> I've hit a bit of a burnout phase. I'll come back to this project in the future

feel free not to.
why not make something actually fun instead? like that one mech pr you had planned ages ago

**Pull request closed without merging**